### DISCUSSION TITLE
Folder USER/src - can be deleted safely?

### DISCUSSION THREAD

**Discussion by xdeama** [postmodern/ruby-install#436] [Q&A] [answered]
Hi ruby-install experts,

I've installed ruby 3.1.2 with install-ruby via homebrew. The installation failed twice just as described here: https://github.com/postmodern/ruby-install/issues/430, before I found a solution (ruby-install ruby -- --enable-shared, it's a M1 Mac).

There's a "src" folder now in my user folder with an unpacked ruby source and a ruby source tar.

/Users/<user>/src/ruby-3.1.2/..
/Users/<user>/src/ruby-3.1.2.tar.xz

Can I safely delete those or is there a built in way to clean up the presumably temporary files?

Best regards
xdeama

**Comment by postmodern** [comment, marked as answer]
You technically can delete them, however there are reasons why it's a good idea to keep it around.

1. It allows re-installing without re-downloading.
2. It leaves a papertrail for how you installed the ruby version.
3. It allows you to apply one-off patches and re-compile/re-install; might be necessary for an emergency security release or if you want to mess around with patches.
4. If you ever need to debug a `ruby` segfault using `gdb`, `gdb` can print the actual lines of the source code where the crash is occurring by reading them from the source files. If the source code which ruby was compiled from is deleted, `gdb` will only print the filename/line number.

**Reply by xdeama** [reply]
many thanks for the quick reply!
